src/Illuminate/Collections/Collection.php

```diff
@@ -1758,4 +1758,23 @@ public function offsetUnset($key): void
     {
         unset($this->items[$key]);
     }
+
+    /**
+     * Calculates the percentage of items that pass a given truth test.
+     *
+     * @param  (callable(TValue, TKey): bool)  $callback
+     * @param  int  $precision
+     * @return float
+     */
+    public function percentage(callable $callback, int $precision = 2): float
+    {
+        if ($this->isEmpty()) {
+            return 0;
```

Comment:
In projects I've worked on, I always have argued in favor of **not** returning 0 from functions like this because the computation never ran. I would normally return null to indicate that this is an exceptional case. That way I have a distinction between 0/10 meet the callback, rather than 0/0 meet the callback.

I realize it's easy enough to call `$collection->isEmpty()` before running this, and ultimately you have to check afterwards if the value is null or a proper float. 🤷

Reply:
Oh, that's an interesting point, because it could have some cases where the percentage itself could be 0. I'll update it to return null, thank you so much! 💪